lib/cli-engine/cli-engine.js

```diff
@@ -897,7 +901,7 @@ class CLIEngine {
         // Clear the last used config arrays.
         lastConfigArrays.length = 0;
         if (resolvedFilename && this.isPathIgnored(resolvedFilename)) {
-            if (warnIgnored) {
+            if (warnIgnored ?? engineWarnIgnored) {
```

Comment:
Using nullish coalescing here because `engineWarnIgnored` should only be used when `warnIgnored` is null or undefined